### ISSUE TITLE
sensor.boiler_auxelecheatnrgconsheating has unknown and NULL since upgrade to 3.5.0b12

### ISSUE THREAD

**Issue by HansRemmerswaal** [emsesp/EMS-ESP32#834]
**Bug description**
Since the upgrade to 3.5.0b12 the sensor.boiler_auxelecheatnrgconsheating (and also the longname version) contains NULL and unknown. This was not the case before the ugrade.

**Screenshots**
![image](https://user-images.githubusercontent.com/116959990/209432448-3cedca4d-bdad-4ab3-97bd-4ec32c271d99.png)

**Device information**
[emsesp_info (1).txt](https://github.com/emsesp/EMS-ESP32/files/10298398/emsesp_info.1.txt)

**Comment by proddy** [comment]
is this in the MQTT topic, the console, web, HA, ioBroker? Can you be more specific

**Comment by HansRemmerswaal** [comment]
This is when I query the SQLite database in HA.

SELECT *
FROM "states"
where entity_id in ('sensor.boiler_auxiliary_electrical_heater_energy_cons','sensor.boiler_auxiliary_electrical_heater_energy_consumption_heating','sensor.boiler_auxelecheatnrgconsheating')
order by last_updated

**Comment by proddy** [comment]
and do they show values in the console, web UI or in the MQTT topics? It will help me hunt down where the problem is

**Comment by HansRemmerswaal** [comment]
It started since the upgrade to 3.5.0b12, this morning I installed EMS-ESP-3_5_0-dev_14-ESP32 and that corrected the entity_id but still seeing the NULL and unkown.

I see a value in web UI
![image](https://user-images.githubusercontent.com/116959990/209645441-82ca65db-b6d3-4707-9fab-7ea4f4eb7ac6.png)

But maybe because that is the last value being received in HA
![image](https://user-images.githubusercontent.com/116959990/209645616-99d95f31-48c8-4e36-ae00-4f0363c56ae4.png)

Below is the export from HA
[states-export.csv](https://github.com/emsesp/EMS-ESP32/files/10307269/states-export.csv)

**Comment by proddy** [comment]
I think that is just old state information in HA. If you see null or unknowns in the WebUI, Console or MQTT then we have a problem!

**Comment by HansRemmerswaal** [comment]
The last entries are from 20 min ago.
I have deleted all the NULL and unknown from the HA database and will keep an eye on it.

![image](https://user-images.githubusercontent.com/116959990/209668049-c844fe08-9d47-40e0-85d7-9f4290c23235.png)

**Comment by HansRemmerswaal** [comment]
1) From HA database I have deleted from "states" the values NULL and unknown.
2) Restarted EMS-ESP module
3) In HA database I now see the NULL and unknown again

Looking at the timings the values are only NULL and unknown for second, so difficult to see them in the WebUI.
Is there a way that I can capture and share the MQTT messages with you?

![image](https://user-images.githubusercontent.com/116959990/209668997-9741df74-7516-4744-bbfa-6c59f7a4d635.png)

**Comment by proddy** [comment]
the entities are created using MQTT Discovery. Use something like MQTTExplorer and search for the config topic for this 'energy_consumption_heating' and it will point to the boiler_data MQTT topic and which json key it's pulling the data from. Then still using MQTTExplorer watch boiler_data and see if the value appears or disappears. EMS-ESP will only create the HA entities when there is a valid value, and if it not then it will quickly try and remove the HA config topic so its goes from HA. Otherwise HA will keep complaining. 

Also, look at the HA logs. You can increase HA's logging for the MQTT integration which will show you more detailed logs. 

I'll also look into the code to see if something is odd

**Comment by HansRemmerswaal** [comment]
I just installed MQTT Explorer, let's see if I can capture something.

But I wonder...

When I restart the EMS-ESP then I'm getting these NULL and unknown also. In HA I then see the sensor.ems_esp_status disconnected / connected. I see many disconnected / connected, also when I'm not playing with the system. Around the same time I see these NULL and unknown. Maybe there is a link.

From HA:

SELECT *
FROM "states"
where entity_id = 'sensor.ems_esp_status'
order by last_updated

![image](https://user-images.githubusercontent.com/116959990/209693952-64e6d601-8c9c-4b9c-9639-ae0590258943.png)

**Comment by proddy** [comment]
I see null values too when querying the DB, but this may be left over residue from previous versions where the states haven't been cleaned up. For example `ems_esp_status` was renamed to `ems_esp_system_bus_status`.

![image](https://user-images.githubusercontent.com/1230712/209697994-680d4c4e-1563-40f8-8051-affd42898a7b.png)

I'll keep tracking it and see what and why this is happening. Apart from the DB, are you seeing strange readings in HA?

**Comment by HansRemmerswaal** [comment]
> Apart from the DB, are you seeing strange readings in HA?
Apart from this, which started when I upgrade to 3.5.0b12, I do not see strange readings when selecting all devices in HA. All kind of patterns looks similar, some examples shown in the first 2 screenshots.

But having said that, I do notice that for some sensor values it's being split, see screenshot 3-5. Need to dig into this.

![image](https://user-images.githubusercontent.com/116959990/209703333-663a2235-6749-4f6b-8817-6d00e8b568b8.png)
![image](https://user-images.githubusercontent.com/116959990/209703414-53682764-5bd4-4c7c-90f9-f6d0abec8eef.png)

**WIFI strength**
![image](https://user-images.githubusercontent.com/116959990/209703825-b02f873d-4a2f-492d-8b30-79c71dad4e2a.png)

**WIFI RSSI**
![image](https://user-images.githubusercontent.com/116959990/209703889-4d5639d3-caaa-4244-8e1a-6eb187768ae6.png)

**Free memory**
![image](https://user-images.githubusercontent.com/116959990/209703952-80af4427-afb4-40fb-ae06-45314c5bc25c.png)

**Comment by HansRemmerswaal** [comment]
SELECT distinct entity_id
FROM "states"
where entity_id like '%WIFI%'

![image](https://user-images.githubusercontent.com/116959990/209704299-81b5459f-4f5e-43e9-a5d8-a917d5fc0af4.png)

**Comment by proddy** [comment]
to be honest I'm not seeing anything strange in that data you shared from HA. Do you have an example of a sensor that is showing incorrect values in HA (not quering the db).

**Comment by HansRemmerswaal** [comment]
The first 2 screenshots show how I checked on high level for strange values. And all looks ok, same kind of patterns and no values that suddenly stop. 

But I did found 3 values which are "wrong", for those I have raised #850.

With respect to the original issues, unknown and NULL since upgrade to 3.5.0b12. I noticed that the EMS-ESP often lost the connection, maybe due to a restart, and each time it was connected again the unknown and NULL where recorded in the HA database for the sensor values. Not sure if EMS-ESP is sending these or that HA is automatically generating them. But since the last build of EMS-ESP-3_5_0-dev_14-ESP32 I have not seen that lost connection anymore and therefore also not that NULL and unknown anymore (apart from the times that I restarted the EMS-ESP myself).

So it could be that this issue has been resolved as a side effect of the other defects which has been resolved recently.

From my point of view we can close this issue, if I see these restarts again then I can raise a specific issue for that one.
